### ISSUE TITLE
add AUC as metrics for classification problem

### ISSUE THREAD

**Issue by CodeKiller48** [fchollet/keras#4004]
for unbalance( positive and negative samples are not equal sizes) dataset, accuracy may be not a best metrics to evaluate. is that possible implement AUC as binary classification metrics

**Comment by isaacgerg** [comment]
If this happens, can we also add cost curves as well from drummond and holte's paper.

**Comment by LucaCappelletti94** [comment]
Hello, I have implemented a [python package called extra_keras_metrics](https://github.com/LucaCappelletti94/extra_keras_metrics) with all the metrics of TensorFlow brought to Keras. If you guys believe it can be useful I can make a pull request for adding the metrics.
